modules/amazon/index.js

```diff
@@ -0,0 +1,40 @@
+var amazon = require('amazon-product-api');
+
+var amazonClient = null;
+var amazonRegex = /(B[0-9]{2}[0-9A-Z]{7}|[0-9]{9}(?:X|[0-9]))/;
+
+module.exports.init = function(bot) {
+  bot.getConfig("amazon.json", function(err, conf) {
+    if(err) bot.say("Unable to load amazon module: " + err);
+    try {
+      amazonClient = new amazon.createClient(conf);
+    } catch(ex) {
+      bot.say("Error loading amazon library: " + ex);
+    }
+  });
+};
+
+module.exports.commands = [];
+
+module.exports.url = function(url, reply) {
+  if(amazonClient === null) return reply("Unable to handle amazon url; lib not loaded");
+
+  var match;
+  if((match = amazonRegex.exec(url))) {
+    if(match[1]) {
+      var asin = match[0];
+      amazonClient.itemLookup({itemId: asin}, function(err, results) {
+        if(err) reply("Error getting title for ASIN: " + match[0]);
+        else {
+          var itemName = results[0].ItemAttributes[0].Title;
+          var price = results[0].ItemAttributes[0].ListPrice[0].FormattedPrice;
```

Comment:
I would expect this to throw a `TypeError` for something with a ListPrice array of empty (e.g. the case of something that's currently unavailable)

Did you test?